### PR TITLE
Add tests for quests, topics and greeting

### LINKED ISSUE #168: Update quest related tests to include ungraded quests

### PR DESCRIPTION
**Pull request by wilinetan**
Fixes #168

### CODE REVIEW

File: src/main/java/seedu/jarvis/model/topic/Topic.java [hunk 1]
```diff
@@ -18,6 +20,8 @@ public Topic() {
      * @param outline outline of Topic
      */
     public Topic(String week, String outline) {
+        requireNonNull(week);
```
Comment (ngzhenteng) [inline src/main/java/seedu/jarvis/model/topic/Topic.java:23]: Great protective measure!

File: src/test/java/seedu/jarvis/model/greeting/GreetingTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        // When greeting is empty string
+        assertTrue(greeting.isEmpty());
+
+        // When greeting is default message
+        greeting.setGreeting("Hi! Please log in to continue");
+        assertTrue(greeting.isEmpty());
+    }
+
+    @Test
+    void getGreeting() {
```
Comment (ngzhenteng) [inline src/test/java/seedu/jarvis/model/greeting/GreetingTest.java]: Perhaps this test method signature could be setGreeting_validString_success()?

File: src/test/java/seedu/jarvis/model/quest/QuestListTest.java [hunk 3]
```diff
@@ -1,18 +1,75 @@
...
+import seedu.jarvis.testutil.QuestBuilder;
+
 class QuestListTest {
 
+
+    private final QuestList questList = new QuestList();
+    private final List<Quest> questListTest = new ArrayList<>();
+
+    @Test
+    void add_nullQuest_throwsNullPointerException() {
```
Comment (ngzhenteng) [inline src/test/java/seedu/jarvis/model/quest/QuestListTest.java:23]: This type of checking is indeed really important, thank you for writing these tests cases!

**Review by ngzhenteng — APPROVED** [review]
Great additions, thank you! LGTM

File: src/test/java/seedu/jarvis/model/greeting/GreetingTest.java (hunk 2 above)
Comment (wilinetan) [reply src/test/java/seedu/jarvis/model/greeting/GreetingTest.java]: 👍

**Pull request merged**